### ISSUE TITLE
`caplog.set_level()` doesn't override `log_level`

### ISSUE THREAD

**Issue by ruaridhw** [pytest-dev/pytest#7133]
A "global" log level set by `log_level` cannot be overridden at the test level.

Minimal Example:

```python
# setup.cfg
[tool:pytest]
log_level = CRITICAL

# test_example.py
import logging


def function_that_logs():
    logging.debug('message')


def test_logger(caplog):
    caplog.set_level(logging.DEBUG)
    function_that_logs()
    assert caplog.text
```

Running `pytest-master@077d1c3`, Python 3.7.6.

It looks like this is because `log_level` adjusts the level of `LogCaptureHandler()` in `_runtest_for_main()` whereas `caplog.set_level()` adjusts the level of a specific `Logger` (likely `Root`).

If I add `self.handler.setLevel(level)` to the end of `LogCaptureFixture.set_level()` the above example passes. Is this a suitable fix (assuming the corresponding cleanup is added too)? Or is this a misuse of `log_level`?

**Comment by thisch** [comment]
Thx for reporting this issue.

`LogCaptureFixture` doesn't have a `handler` attribute. So what do `self` in your proposed fix? You could run the unit tests of pytest locally (or via CI) to check that you don't break anything.

**Comment by ruaridhw** [comment]
> So what do `self` in your proposed fix?

The `handler` property of `LogCaptureFixture`.

**Comment by thisch** [comment]
Sry I overlooked that this class has a custom property called `handler`. 

I guess that setting the level of a test-local `LogCaptureHandler` (returned by `LogCaptureFixture.handler`) will have unwanted side effects, but I'm not 100% sure. 

Do the tests pass locally?

**Comment by ruaridhw** [comment]
>will have unwanted side effects

Yes, I imagine that any tests that *rely* on the `ini` being a particular level and then assert that no messages were emitted without calling `set_level()` themselves, will now fail.

**Comment by bluetech** [comment]
@ruaridhw Which of the following is your use case?

1. Change which log messages are captured by `caplog`.
2. Change which log messages are displayed in the test reports.

I assume it's 1, since 2 is not really something a test should be able to adjust, but just wanted to make sure.

If it's 1, then this should be resolved if #7162 is resolved (without needing #7139).

**Comment by ruaridhw** [comment]
> I assume it's 1, since 2 is not really something a test should be able to adjust, but just wanted to make sure.

Yes, you're right, it's 1.

**Comment by nicoddemus** [comment]
Fixed in #7340
